src/get-jwks.d.ts

```diff
@@ -1,4 +1,5 @@
 import LRU from 'lru-cache'
+import type { Agent } from 'https'
```

Comment:
is http and https agent the same type?

Reply:
`https.Agent` extends `http.Agent` ([source](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v16/https.d.ts#L17-L28)), so the typing as I've declared it will reject plain HTTP agents:
```ts
    // https.d.ts

    interface AgentOptions extends http.AgentOptions, tls.ConnectionOptions {
        rejectUnauthorized?: boolean | undefined;
        maxCachedSessions?: number | undefined;
    }
    /**
     * An `Agent` object for HTTPS similar to `http.Agent`. See {@link request} for more information.
     * @since v0.4.5
     */
    class Agent extends http.Agent {
        constructor(options?: AgentOptions);
        options: AgentOptions;
    }
```

Reply:
I agree with your reasons, I was just asking if this was intentional or casual